localMatchApp/sources/Login.cpp

```diff
@@ -0,0 +1,82 @@
+#include "Login.h"
+#include <iostream>
+
+using namespace std;
+
+Login::Login ( ) {
+  // ToDo: Connect to database.
+  // This console out is created for debug process
+  std::cout<<"object created"<<std::endl;
+}
+
+Login::~Login ( ) {
+  // ToDo: Disconnect to database.
+  // This console out is created for debug process
+  std::cout<<"object deleted"<<std::endl;
+}
+
+Login::ConnectionMessage Login::signUp ( const string& i_user, const string& i_password,
+                                         const string& i_repeatedPassword ) {
+  ConnectionMessage signUpConnectionState=ConnectionMessage::goodEntry;
+  // ToDo: Send and recive information to database.
+  // This string is just created for debug process
+  string userInDatabase = "Lol2018";
+
+  if ( !hasCorrectFormat ( i_password ) || !hasCorrectFormat ( i_repeatedPassword ) ) {
+    signUpConnectionState= ConnectionMessage::wrongPassword;
+  }
+
+  if ( i_password != i_repeatedPassword && (hasCorrectFormat ( i_password ) && hasCorrectFormat ( i_repeatedPassword ))) {
+    signUpConnectionState=ConnectionMessage::notEqualPasswords;
+  }
+
+  if ( i_user==userInDatabase || i_user.length()==0) {
+    signUpConnectionState= ConnectionMessage::invalidUser;
+  }
+
+  // ToDo: Set user and password in table.
+  return signUpConnectionState;
+}
+
+Login::ConnectionMessage Login::entry ( const string& i_user, const string& i_password ) {
+  ConnectionMessage signUpConnectionState=ConnectionMessage::goodEntry;
+  // ToDo: Search the user and compare password
+  // ToDo: Query asking for password based on user
+  // These strings are just created for debug process
+  string userInDatabase = "Lol2018";
+  string passInDatabase = "Lol2018";
+
+  if ( i_user != userInDatabase ) {
+    signUpConnectionState= ConnectionMessage::doesntExists;
+  }
+  if ( i_password != passInDatabase && (i_user == userInDatabase) ) {
+    signUpConnectionState= ConnectionMessage::wrongPassword;
+  }
+  return signUpConnectionState;
+}
+
+bool Login::hasCorrectFormat ( const string& i_password ) {
+  bool hasANumber=false;
+  bool hasACapital=false;
+  bool hasASmallLetter=false;
+  bool correctFormat= false;
+
+  for ( const auto &i : i_password ){
+      if ( isupper ( i ) ) {
+        hasACapital= true;
+      }
+
+      if ( islower ( i ) ) {
+        hasASmallLetter= true;
+      }
+
+      if ( isdigit ( i ) ) {
+        hasANumber= true;
+      }
+    }
+
+  if ( hasANumber && hasACapital&& hasASmallLetter) {
```

Comment:
This IF statement could be inside the FOR loop, so you don't have to iterate through all the string to know if the format is valid or not

Reply:
You are right, change applied.